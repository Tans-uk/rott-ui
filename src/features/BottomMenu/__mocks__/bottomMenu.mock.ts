import type {BottomMenuItemModel} from '../models'

export const bottomMenuListItemMock: BottomMenuItemModel[] = [
  {
    testID: 'bottom-menu-item-1-test-id',
    title: 'Menu 1',
    icon: {
      name: 'ARROW_DOWN',
    },
    onPress: jest.fn(),
  },
  {
    testID: 'bottom-menu-item-2-test-id',
    title: 'Menu 2',
    icon: {
      name: 'ARROW_DOWN',
    },
    onPress: jest.fn(),
  },
]
