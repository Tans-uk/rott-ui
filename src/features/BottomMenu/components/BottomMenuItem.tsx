import {type FC} from 'react'

import {Linking} from 'react-native'

import type {BottomMenuItemModel} from '../models'
import {BottomMenuStyles} from '../styles'

import {Icon} from '@features/Icon'
import {Image} from '@features/Image'
import {Label} from '@features/Label'
import {Pressable} from '@features/Pressable'
import {themeConfig} from '@providers'

export const BottomMenuItem: FC<BottomMenuItemModel> = ({
  testID,
  title,
  icon,
  image,
  containerStyle,
  onPress,
  url,
  phone,
}) => {
  const isPhone = !!phone && !phone.isEmpty()

  return (
    <Pressable
      testID={testID}
      alignItemsCenter
      width={containerStyle ? 56 : undefined}
      style={containerStyle}
      onPress={async (event) => {
        if (process.env.NODE_ENV === 'development' && isPhone) return
        else if (isPhone) await Linking.openURL(`tel:${phone}`)
        else if (url && (await Linking.canOpenURL(url))) await Linking.openURL(url)
        else if (onPress) onPress(event)
      }}>
      {icon && !image && (
        <Icon
          {...icon}
          name={icon.name}
          width={icon?.width ? icon.width : 24}
          height={icon?.height ? icon.height : 24}
          variant={icon.variant ?? 'grey-900'}
          mode={icon?.mode}
          strokeWidth={icon?.strokeWidth ? icon?.strokeWidth : 2}
          marginBottom={8}
        />
      )}

      {image && !icon && (
        <Image
          {...image}
          width={image?.width ? image?.width : 24}
          height={image?.height ? image.height : 24}
          tintColor={image?.width && image?.height ? undefined : themeConfig.colors['grey-900']}
          marginBottom={8}
          resizeMode='contain'
          name={image.name}
        />
      )}

      {title && (
        <Label
          style={BottomMenuStyles().textStyle}
          variant='grey-900'
          fontSize='xs'
          fontWeight='bold'
          numberOfLines={1}>
          {title}
        </Label>
      )}
    </Pressable>
  )
}
