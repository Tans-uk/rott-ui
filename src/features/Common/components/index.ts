export * from './CommonItem'
export * from './CommonItemContainer'
