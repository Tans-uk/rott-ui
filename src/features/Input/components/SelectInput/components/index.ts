export * from './SelectInput'
export * from './SelectInputModalComponent'
