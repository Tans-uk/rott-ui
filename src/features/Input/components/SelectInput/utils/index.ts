export * from './listHeightNormalizer'
export * from './modalHeightPercentageNormalizer'
export * from './sortListBySearchPriority'
