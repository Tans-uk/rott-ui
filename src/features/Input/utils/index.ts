export * from './formatByDateMode'
export * from './useInputStyleNormalizer'
