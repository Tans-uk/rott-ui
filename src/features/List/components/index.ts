export * from './List'
export * from './FavoriteSwipeComponent'
export * from './ListSkeletonItem'
