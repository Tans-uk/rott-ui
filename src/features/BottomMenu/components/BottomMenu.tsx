import React, {type FC} from 'react'

import {Platform} from 'react-native'

import {useSafeArea} from '../../../hooks'
import {themeConfig} from '../../../providers'
import {Content} from '../../Content'
import type {BottomMenuItemModel} from '../models'
import {BottomMenuItem} from './BottomMenuItem'

interface BottomMenuProps {
  menuItems: BottomMenuItemModel[]
}

export const BottomMenu: FC<BottomMenuProps> = ({menuItems}) => {
  const {bottom} = useSafeArea()

  let absoluteHeight = 72 + bottom // 56 dip yükseklik + 10 dip boşluk + bottom inset
  if (Platform.OS === 'ios' && bottom > 0) absoluteHeight -= 16 // ios'te navigation bar şeffaf ve boşluklu olduğu için 16 dip boşluğa gerek yoktur

  return (
    <Content
      testID='bottom-menu-container'
      row
      absolute
      noPadding
      bottom={0}
      paddingHorizontal={8}
      size='full'
      borderTopStartRadius={24}
      borderTopEndRadius={24}
      justifyContentSpaceAround
      backgroundColor={themeConfig.colors.secondary}
      height={absoluteHeight}
      heightNormalizeBased>
      {menuItems.map((props) => (
        <BottomMenuItem key={`${props?.title ?? 'Main_'}`} {...props} />
      ))}
    </Content>
  )
}
