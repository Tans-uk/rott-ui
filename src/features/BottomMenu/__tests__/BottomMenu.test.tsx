import {bottomMenuListItemMock} from '../__mocks__'
import {BottomMenu} from '../components'

import {render} from 'src/__tests__/utils/testUtils'

describe('Component -> BottomMenu', () => {
  const testIDs = {
    bottomMenuContainerTestId: 'bottom-menu-container',
    bottomMenuItemTestId: 'bottom-menu-item-1-test-id',
    bottomMenuItemTestId2: 'bottom-menu-item-2-test-id',
  }

  it('bottom menu componenti snapshotı ile eşleşmeli', () => {
    const rendered = render(<BottomMenu menuItems={bottomMenuListItemMock} />)

    expect(rendered).toMatchSnapshot()
  })

  it('ekranda görülmeli', () => {
    const {getByTestId} = render(<BottomMenu menuItems={bottomMenuListItemMock} />)
    const bottomMenuContainer = getByTestId('bottom-menu-container')

    expect(bottomMenuContainer).toBeOnTheScreen()
  })

  it('bottom menu componenti ekranda gözükmeli', () => {
    const {bottomMenuContainerTestId} = testIDs

    const {getByTestId} = render(<BottomMenu menuItems={bottomMenuListItemMock} />)

    const bottomMenu = getByTestId(bottomMenuContainerTestId)

    expect(bottomMenu).toBeOnTheScreen()
  })

  it('verilen menu item sayısı kadar bottom menu item componenti ekranda gözükmeli', () => {
    const {bottomMenuItemTestId, bottomMenuItemTestId2} = testIDs

    const {getByTestId} = render(<BottomMenu menuItems={bottomMenuListItemMock} />)

    const bottomMenuItem = getByTestId(bottomMenuItemTestId)
    const bottomMenuItem2 = getByTestId(bottomMenuItemTestId2)

    expect(bottomMenuItem).toBeOnTheScreen()
    expect(bottomMenuItem2).toBeOnTheScreen()
  })
})
