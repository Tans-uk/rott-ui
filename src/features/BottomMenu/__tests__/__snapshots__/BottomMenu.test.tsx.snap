// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component -> BottomMenu bottom menu componenti snapshotı ile eşleşmeli 1`] = `
<View
  flex={1}
  pointerEvents="box-none"
  style={
    {
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    absolute={true}
    borderTopEndRadius={24}
    borderTopStartRadius={24}
    bottom={0}
    height={72}
    heightNormalizeBased={true}
    style={
      {
        "backgroundColor": "rgba(255, 199, 44, 1)",
        "borderTopEndRadius": 24,
        "borderTopStartRadius": 24,
        "bottom": 0,
        "flex": undefined,
        "flexDirection": "row",
        "height": 114,
        "justifyContent": "space-around",
        "maxHeight": 114,
        "paddingBottom": 0,
        "paddingHorizontal": 13,
        "position": "absolute",
        "width": "100%",
      }
    }
    testID="bottom-menu-container"
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      alignItemsCenter={true}
      animated={false}
      collapsable={false}
      focusable={true}
      height={56}
      justifyContentCenter={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "height": 108,
          "justifyContent": "center",
          "maxHeight": 108,
        }
      }
      testID="bottom-menu-item-1-test-id"
    >
      <View
        marginBottom={4}
        style={
          {
            "flex": undefined,
            "flexDirection": "column",
            "marginBottom": 8,
          }
        }
      >
        <Text
          fill="rgba(34, 63, 70, 1)"
          height={46}
          strokeWidth={2}
          testID="mock-icon"
          width={46}
        >
          MockIcon
        </Text>
      </View>
      <Text
        numberOfLines={1}
        style={
          {
            "color": "rgba(34, 63, 70, 1)",
            "fontFamily": "Markpro-Bold",
            "fontSize": 10,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Menu 1
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      alignItemsCenter={true}
      animated={false}
      collapsable={false}
      focusable={true}
      height={56}
      justifyContentCenter={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "height": 108,
          "justifyContent": "center",
          "maxHeight": 108,
        }
      }
      testID="bottom-menu-item-2-test-id"
    >
      <View
        marginBottom={4}
        style={
          {
            "flex": undefined,
            "flexDirection": "column",
            "marginBottom": 8,
          }
        }
      >
        <Text
          fill="rgba(34, 63, 70, 1)"
          height={46}
          strokeWidth={2}
          testID="mock-icon"
          width={46}
        >
          MockIcon
        </Text>
      </View>
      <Text
        numberOfLines={1}
        style={
          {
            "color": "rgba(34, 63, 70, 1)",
            "fontFamily": "Markpro-Bold",
            "fontSize": 10,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Menu 2
      </Text>
    </View>
  </View>
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      [
        {
          "alignSelf": "center",
          "elevation": 999999,
          "flex": 0,
          "maxWidth": "100%",
          "position": "absolute",
          "zIndex": 999999,
        },
        {
          "flexDirection": "column-reverse",
          "justifyContent": "flex-start",
          "top": 30,
          "width": 750,
        },
      ]
    }
  >
    <View
      style={
        {
          "bottom": 0,
        }
      }
    >
      <RCTSafeAreaView />
    </View>
  </View>
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      [
        {
          "alignSelf": "center",
          "elevation": 999999,
          "flex": 0,
          "maxWidth": "100%",
          "position": "absolute",
          "zIndex": 999999,
        },
        {
          "bottom": 40,
          "flexDirection": "column",
          "justifyContent": "flex-end",
          "width": 750,
        },
      ]
    }
  >
    <View
      style={
        {
          "bottom": 0,
        }
      }
    >
      <RCTSafeAreaView />
    </View>
  </View>
</View>
`;
